src/interval.rs

```diff
@@ -15,7 +15,14 @@ const BASE: usize = 2;
 
 /// Binary representation of a price interval.
 #[derive(Clone, Debug, PartialEq)]
-pub struct Digits(BitVec);
+pub struct Digits {
+    inner: BitVec,
+
+    /// The precomputed range represents by these `Digits`.
+    ///
+    /// This is pre-computed for performance reasons.
+    range: RangeInclusive<u64>,
```

Comment:
👍 I suspected you would end up doing this when I looked at the code a few hours ago.